### PR TITLE
[mir] Drop platform

### PR DESCRIPTION
**Pull request by hudokkow**
## Description
Drop MIR platform.

## Motivation and Context
Doesn't build on 17.10 or 18.04.

## How Has This Been Tested?
Brief runtime test on Ubuntu.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Cosmetic change (non-breaking change that doesn't touch code)
- [X] None of the above (please explain below)

#### Clean up

cc/ @BrandonSchaefer

### CODE REVIEW

**Comment by fritsch** [comment]
Did you talk to @BrandonSchaefer before? I still think Mir has some support by canonical: https://blog.ubuntu.com/2018/07/10/graphical-environments-in-the-world-of-iot?_ga=2.216416790.1242921367.1531492440-211165890.1454419801

**Comment by hudokkow** [comment]
I haven't. This is only to get his attention. ;)

**Comment by BrandonSchaefer** [comment]
Wayland clients work on Mir (Though dont think anyone has tested Kodi Wayland on Mir yet). So with the Wayland platform there is no need for this backend.

+1 from me

**Comment by fritsch** [comment]
@FernetMenta okay?

@BrandonSchaefer thx very much for your reply!

**Comment by FernetMenta** [comment]
sure

**Comment by lrusak** [comment]
:+1:

**Comment by hudokkow** [comment]
@BrandonSchaefer, thank you for clarifying!

File: cmake/platform/linux/mir.cmake [hunk 1]
```diff
@@ -1,3 +1,3 @@
-set(PLATFORM_REQUIRED_DEPS OpenGl EGL Mir LibDRM)
+set(PLATFORM_REQUIRED_DEPS OpenGl EGL LibDRM)
```
Comment (Rechi) [inline cmake/platform/linux/mir.cmake]: Why was this file not removed?

Comment (hudokkow) [reply cmake/platform/linux/mir.cmake]: Oops! I'll PR the removal

**Pull request merged**